### ISSUE TITLE
Remove validation doesn't work if you try twice

### ISSUE THREAD

**Issue by nettino** [ghiscoding/angular-validation#91]
http://plnkr.co/edit/6J0ZApXkOKUB2T0tz8iZ
Way to reproduce:
Click "add validation" type in, click "remove validation" type - repeat twice then start typing again.

**Comment by nettino** [comment]
or is there any better way to make dependant fields (ie. if checkbox ticked, validate other field, else don't validate)?

**Comment by ghiscoding** [comment]
Removing and Adding back a Validator should not be used that way. I only made those for flexibility purposes but I strongly suggest to use something else instead.

With your checkbox example, I would use interpolation to do it, it's simple to code and it should work all the time. Take a look at [\more-examples\interpolateValidation](https://github.com/ghiscoding/angular-validation/tree/master/more-examples/interpolateValidation), that example is with a checkbox as you want.

**Comment by nettino** [comment]
Hi - thanks for that! I found one bug - for more complex validation the message won't appear once the field it's dirty (it will only appear after any interaction...). I've just replaced 
```javascript
return vm.validationRequired ? 'required' : '';
```
with 
```javascript
return vm.validationRequired ? 'alpha_dash|min_len:3|max_len:50|required' : '';
```
in your example.

**Comment by nettino** [comment]
Please fix.

**Comment by ghiscoding** [comment]
You know this is Open Source Community, you should try to help instead of just asking. I do this on my free time and I'm not getting paid at all for this, so be patient or help.

**Comment by nettino** [comment]
I've tried to take a look, but there's too much code for me ;) I'm just wondering why aren't you registering your validators with the default Angular validation, but create your own? - I think it's for pre-1.3 compatibility? Correct me if I'm wrong. 
Also - I've started to modify your plugin to get even better user experience (I'm not saying it's bad!) - I mean once the user types the error comes out only once he finishes (and it stays there until next update), but... once the field becomes valid it's marked valid immediately! - and I've done it for my config, but probably with other settings it will have some issues and I need to pass if the blur event occured from blurHandler down to updateErrorMsg function which looks a bit complex. Also I am not sure what to observe with my decorators...
I am probably not grasping the whole concept - it would be great if you shared some consized description how the system works otherwise we may end up with reading a hundreds lines of code and run out of time to help... :(
Anyway I do appreciate the plugin - there must have been a lot of work put into it!

**Comment by ghiscoding** [comment]
I started the project when 1.2.x was default, it still use the same validation as Angular, both of them will give you `$error` property and class like `ng-invalid` when invalid. 
Since it's 1.2.x, you can imagine that It's been a long time and more people started using it, then more features people started to ask for. But as I said, this is all done on my free time, so there's only so much I can do, I still have a life at the end. Changing the way my Angular-Validation works is all up to you, I defined it the way that I thought would be the best for my taste but we all have different tastes. Even though, my package is quite versatile in functionalities and options, so there's still ways to make it to your taste as you did with decorators.

In your open issue, it seems that the interpolation is not reinterpreted, so it seems to be a big problem and might take a while to troubleshoot all that.

**Comment by ghiscoding** [comment]
This was a watcher problem, after unbinding the validation on the first time, it was not putting back the `$watch` properly. I'm refactoring some of my code though it looks like I have something working, I will need to run all my Protractor tests and will probably push by the weekend end.

**Comment by ghiscoding** [comment]
Fixed in latest release [v1.4.18](https://github.com/ghiscoding/angular-validation/releases/tag/v1.4.18)

Thanks for the feedback and patience.
